api-server/models.py

```diff
@@ -17,15 +24,133 @@ class State(str, Enum):
+                    author=self,
+                )
+            )
+
+        await Repository.bulk_create(repo_models)
+        logger.info(
+            f"Found {len(repo_models)} available repos for author {self.username}"
+        )
+
+    async def load_files(self, gh_client: GithubClient):
+        async with in_transaction():
+            repos = await self.repos.filter(load_status=False).order_by(
+                "last_pushed_at"
+            )
+
+            loaded_repo_ids = []
+            repo_index = 0
+            non_empty_repo_count = 0
+            file_models = []
+            while (
+                non_empty_repo_count < self.MAX_REPOS_TO_GENERATE_FILES
+                and repo_index < len(repos)
+            ):
+                repo = repos[repo_index]
+                try:
+                    files = await gh_client.get_files_for_repo(
+                        repo.name,
+                        repo.default_branch,
+                        self.SUPPORTED_EXTENSIONS,
+                    )
+                    for file in files:
+                        file_models.append(
+                            File(
+                                name=file["name"],
+                                path=file["path"],
+                                download_url=file["download_url"],
+                                repo=repo,
+                            )
+                        )
+                    loaded_repo_ids.append(repo.id)
+                    repo_index += 1
+                    if len(files) > 0:
+                        non_empty_repo_count += 1
+                except GithubRepositoryFileLoadingError:
+                    pass
+
+            if len(loaded_repo_ids) > 0:
+                await Repository.filter(id__in=loaded_repo_ids).update(load_status=True)
+            if len(file_models) > 0:
+                await File.bulk_create(file_models)
+            logger.info(
+                f"Extracted {len(file_models)} files from {len(loaded_repo_ids)} repos for author {self.username}"
+            )
+
+
+class Repository(models.Model):
+    id = fields.UUIDField(pk=True)
+
+    load_status = fields.BooleanField(default=False)
+    name = fields.CharField(max_length=100)
+    description = fields.CharField(max_length=500, null=True)
+    stars = fields.IntField()
+    language = fields.CharField(max_length=20, null=True)
+    default_branch = fields.CharField(max_length=100)
+    url = fields.CharField(max_length=200)
+    last_pushed_at = fields.DatetimeField()
+
+    author: fields.ForeignKeyRelation[Session] = fields.ForeignKeyField(
+        "models.Player", "repos"
+    )
+
+
+class File(models.Model):
+    id = fields.UUIDField(pk=True)
+
+    name = fields.CharField(max_length=100)
+    path = fields.CharField(max_length=200)
+    download_url = fields.CharField(max_length=200)
+
+    repo: fields.ForeignKeyRelation[Repository] = fields.ForeignKeyField(
+        "models.Repository", "files"
```

Comment:
again! so we can just use `files` on a `Player` to access all the files?

Reply:
Yep. In this case, it would be `Repository.files`.